ILSpy.AddIn.Shared/ILSpyInstance.cs

```diff
@@ -47,85 +44,28 @@ public void Start()
 		{
 			var commandLineArguments = parameters?.AssemblyFileNames?.Concat(parameters.Arguments);
 			string ilSpyExe = GetILSpyPath();
-			var process = new Process() {
-				StartInfo = new ProcessStartInfo() {
-					FileName = ilSpyExe,
-					UseShellExecute = false,
-					Arguments = "/navigateTo:none"
-				}
-			};
-			process.Start();
+
+			const string defaultOptions = "--navigateto:none";
+			string argumentsToPass = defaultOptions;
 
 			if ((commandLineArguments != null) && commandLineArguments.Any())
 			{
-				// Only need a message to started process if there are any parameters to pass
-				SendMessage(ilSpyExe, "ILSpy:\r\n" + string.Join(Environment.NewLine, commandLineArguments), true);
-			}
-		}
-
-		[System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "VSTHRD110:Observe result of async calls", Justification = "<Pending>")]
-		void SendMessage(string ilSpyExe, string message, bool activate)
-		{
-			string expectedProcessName = Path.GetFileNameWithoutExtension(ilSpyExe);
-			// We wait asynchronously until target window can be found and try to find it multiple times
-			Task.Run(async () => {
-				bool success = false;
-				int remainingAttempts = 20;
-				do
-				{
-					NativeMethods.EnumWindows(
-						(hWnd, lParam) => {
-							string windowTitle = NativeMethods.GetWindowText(hWnd, 100);
-							if (windowTitle.StartsWith("ILSpy", StringComparison.Ordinal))
-							{
-								string processName = NativeMethods.GetProcessNameFromWindow(hWnd);
-								Debug.WriteLine("Found {0:x4}: '{1}' in '{2}'", hWnd, windowTitle, processName);
-								if (string.Equals(processName, expectedProcessName, StringComparison.OrdinalIgnoreCase))
-								{
-									IntPtr result = Send(hWnd, message);
-									Debug.WriteLine("WM_COPYDATA result: {0:x8}", result);
-									if (result == (IntPtr)1)
-									{
-										if (activate)
-											NativeMethods.SetForegroundWindow(hWnd);
-										success = true;
-										return false; // stop enumeration
-									}
-								}
-							}
-							return true; // continue enumeration
-						}, IntPtr.Zero);
+				string assemblyArguments = string.Join(" ", commandLineArguments);
 
-					// Wait some time before next attempt
-					await Task.Delay(500);
-					remainingAttempts--;
-				} while (!success && (remainingAttempts > 0));
-			});
-		}
+				string filepath = Path.GetTempFileName();
```

Comment:
who is going to delete the file?

Reply:
Resolved via https://github.com/icsharpcode/ILSpy/pull/3205/commits/d2221783ba211be78072701876a1ed490248230c